lib/pure/strutils.nim

```diff
@@ -1245,11 +1293,7 @@ proc parseEnum*[T: enum](s: string): T =
     doAssert parseEnum[MyEnum]("second") == second
     doAssertRaises(ValueError):
       echo parseEnum[MyEnum]("third")
-
-  for e in low(T)..high(T):
-    if cmpIgnoreStyle(s, $e) == 0:
-      return e
-  raise newException(ValueError, "invalid enum value: " & s)
+  genEnumStmt(T, s, default = false)
```

Comment:
bool type is also enum, likely this example will trigger incorrect behavior:
```nim
parseEnum[bool]("xx", false) # exception instead of `false`
```
I suggest as default to use `nil` value

Reply:
Good catch, I forgot that that's how bools work.